recipes/opencv/4.x/conanfile.py

```diff
@@ -284,6 +298,19 @@ def add_components(components):
                 if self.settings.os == "Linux":
                     self.cpp_info.components[conan_component].system_libs = ["dl", "m", "pthread", "rt"]
 
+                if self.settings.os == "Android":
+                    self.cpp_info.components[conan_component].includedirs = [
+                        os.path.join("sdk", "native", "jni", "include")]
+                    self.cpp_info.components[conan_component].system_libs.append("log")
+                    # self.cpp_info.components[conan_component].system_libs.append("mediandk")  # available for ANDROID_NATIVE_API_LEVEL > 20
+                    if not self.options.shared:
+                        self.cpp_info.components[conan_component].libdirs.append(
+                            os.path.join("sdk", "native", "staticlibs", tools.to_android_abi(str(self.settings.arch))))
+                        if conan_component == "opencv_core":
+                            self.cpp_info.components[conan_component].libdirs.append("lib")
+                            self.cpp_info.components[conan_component].libs.append("cpufeatures")  # always available
+                            # self.cpp_info.components[conan_component].libs.append("tegra_hal")  # available for ARM OR AARCH64 but not IOS
```

Comment:
you can check `self.settings.arch` and `self.settings.os`

Reply:
Hm, this one is more complicated I guess, and I'm afraid this is a little bit too much for this first step, supporting Android. the rule inside OpenCV's CMakeLists.txt is:

```cmake
OCV_OPTION(WITH_CAROTENE "Use NVidia carotene acceleration library for ARM platform" ON
  VISIBLE_IF (ARM OR AARCH64) AND NOT IOS AND NOT (CMAKE_VERSION VERSION_LESS "2.8.11"))
```

There is a condition for ARM, AARCH64, IOS and CMake Version... Trying to model this in the recipe looks like more code than just one line or two. A more simpler approach would be: can we check if a library `tegra_hal.a` exists in the `lib` folder and add it if it is available? Then we could avoid to rebuild the internal rule (which may change in future versions).

At the bottom line, if we cannot find a simple way to add this conditionally, I would to prefer to disable it and let it open for later improvements, when more other platforms will be added. 

What do you think?

Reply:
I think I found the solution, using tools.collect_libs(). This removes all the rule reproduction, is shorter and clean.

Reply:
still, conditional might be needed for the options above?
but if it's too complicated, it's okay to omit for the first iteration, so up to you

Reply:
I'm just waiting for the current check to pass, then I push my next version. For `libmediandk.so`, the condition is still required, yes, but this is simple enough (only NDK API level dependency), and others go away completely. I hope this will be the best solution then.